### PR TITLE
Added missing documentation for media.js

### LINKED ISSUE #7: Document wp-admin/js/media.js

### PR DESCRIPTION
**Pull request by jcomack**
Fixes #7 

Worked on this together with @diedexx.

### CODE REVIEW

File: src/wp-admin/js/media.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
 	findPosts = {
+		/**
+		 * @summary Opens a dialog to attach media to a post.
+		 *
+		 * Adds an overlay prior to retrieving a list of posts to attach the media to.
+		 *
+		 * @param {string} af_name The name of the affected element.
+		 * @param {string} af_val The value of the affected post element.
+		 * @returns {boolean} Always returns false.
+		 */
```
Comment (rensw90) [inline src/wp-admin/js/media.js]: Missing ```@since```

File: src/wp-admin/js/media.js [hunk 2]
```diff
@@ -1,8 +1,27 @@
 /* global ajaxurl, attachMediaBoxL10n, _wpMediaGridSettings, showNotice */
 
+/**
+ * Creates a dialog and retrieves all the posts that a particular media item can be attached to.
+ *
+ * @summary Creates a dialog containing posts that can have a particular media attached to it.
```
Comment (rensw90) [inline src/wp-admin/js/media.js]: You don't to add both a summary and a long description if the long description is just one line.

File: src/wp-admin/js/media.js [hunk 3]
```diff
@@ -14,35 +40,69 @@ var findPosts;
...
+			 */
 			$('#find-posts-input').focus().keyup( function( event ){
 				if ( event.which == 27 ) {
 					findPosts.close();
-				} // close on Escape
+				}
 			});
 
-			// Pull some results up by default
+			// Show list of applicable posts for media attachment.
```
Comment (andizer) [inline src/wp-admin/js/media.js]: Shouldn't this be sending something?

File: src/wp-admin/js/media.js [hunk 4]
```diff
@@ -14,35 +40,69 @@ var findPosts;
...
 			findPosts.send();
 
 			return false;
 		},
 
+		/**
+		 * @summary Clears the found posts lists before hiding the attach media dialog.
+		 *
+		 * @since 2.7
```
Comment (andizer) [inline src/wp-admin/js/media.js]: semver

File: src/wp-admin/js/media.js [hunk 5]
```diff
@@ -14,35 +40,69 @@ var findPosts;
...
 			$( '.ui-find-overlay' ).on( 'click', function () {
 				findPosts.close();
 			});
 		},
 
+		/**
+		 * @summary Retrieves and displays posts based on the search term.
+		 *
+		 * Sends a post request to the admin_ajax.php requesting posts based on the search term provided by the user.
+		 * Defaults to all posts if no search term is provided.
```
Comment (andizer) [inline src/wp-admin/js/media.js]: For readability, please add a comma after `admin_ajax.php`

File: src/wp-admin/js/media.js [hunk 6]
```diff
@@ -85,19 +154,30 @@ var findPosts;
...
+
+		// When hitting the enter key in the search input, submit the search query.
 		$( '#find-posts .find-box-search :input' ).keypress( function( event ) {
 			if ( 13 == event.which ) {
 				findPosts.send();
 				return false;
 			}
 		});
+
+		// Bind the search button on click event.
```
Comment (andizer) [inline src/wp-admin/js/media.js]: Binds the click event to the search button?

File: src/wp-admin/js/media.js [hunk 7]
```diff
@@ -85,19 +154,30 @@ var findPosts;
...
 		$( '#find-posts-search' ).click( findPosts.send );
+
+		// Bind the close dialog click event.
```
Comment (andizer) [inline src/wp-admin/js/media.js]: What is binding to what?

File: src/wp-admin/js/media.js [hunk 8]
```diff
@@ -112,7 +192,11 @@ var findPosts;
 			});
 		});
 
-		// Enable whole row to be clicked
+		/**
+		 * @summary Enables a whole row to be clickable inside the post finder.
```
Comment (andizer) [inline src/wp-admin/js/media.js]: What is whole row? Can you be more specific?

**Comment by moorscode** [comment]
Patch created https://core.trac.wordpress.org/ticket/41072

**Comment by omarreiss** [comment]
patch accepted

**Pull request closed without merging**